helix-term/src/commands.rs

```diff
@@ -2659,9 +2661,68 @@ fn last_picker(cx: &mut Context) {
 }
 
 // I inserts at the first nonwhitespace character of each line with a selection
+// If the line is empty, automatically indent
 fn insert_at_line_start(cx: &mut Context) {
-    goto_first_nonwhitespace(cx);
     enter_insert_mode(cx);
+
+    let (view, doc) = current!(cx.editor);
+
+    let text = doc.text().slice(..);
+    let contents = doc.text();
+    let selection = doc.selection(view.id);
+
+    let language_config = doc.language_config();
+    let syntax = doc.syntax();
+    let tab_width = doc.tab_width();
+
+    let mut ranges = SmallVec::with_capacity(selection.len());
+    let mut offs = 0;
+
+    let mut transaction = Transaction::change_by_selection(contents, selection, |range| {
+        let cursor_line = range.cursor_line(text);
+        let cursor_line_start = text.line_to_char(cursor_line);
+
+        let line_content = line_without_line_ending(&text, cursor_line);
+        if line_content.len_chars() == 0 {
```

Comment:
You are not using `line_content` anywhere except to check if its width it 0. This can be done slightly more efficiently by doing:
```suggestion
        if line_end_char_index(text, cursor_line) == cursor_line_start {
```